src/_pytest/logging.py

```diff
@@ -527,6 +527,11 @@ def __init__(self, config: Config) -> None:
         # File logging.
         self.log_file_level = get_log_level_for_setting(config, "log_file_level")
         log_file = get_option_ini(config, "log_file") or os.devnull
+
+        # Keep the log file relative to the inidir file (if it exists) (#7336)
+        inidir = getattr(getattr(config, "inifile", None), "dirname", None)
```

Comment:
At this point `config.inifile` should always be defined (unless I'm mistaken), so `getattr(config, "inifile", None)` can be replaced with just `config.inifile`.

`inifile` can be None, but instead of using `getattr` on it, I think it is better do use an explicit `is None` check. `getattr()` is using a dynamic feature of Python which is not analyzed and type checked, so I for one prefer to avoid it unless the dynamicism is really needed (e.g. when the name is not a constant).

Finally, I think currently there may be some cases where `inifile` may be an str rather than a `py.path.local`, but that is more of a bug (I'm just working on that as a follow up for the recent TOML PR).

Reply:
Hmm I seem to be mistaken, I thought i was getting a <type str> in some cases; and no 'dirname' attr, but it seems if config.inifile is not None, it is always path.local (as far as our current test suite goes)

Reply:
mypy hates:

```
src\_pytest\logging.py:533: error: "str" has no attribute "dirname"  [attr-defined]
Found 1 error in 1 file (checked 2 source files)
```
I assume some type hinting on the config obj - will check